node/src/components/storage/tests.rs

```diff
@@ -1152,3 +1152,77 @@ fn should_hard_reset() {
     // Test with a hard reset to era 0, deleting all blocks and associated data.
     check(0);
 }
+
+#[test]
+fn can_put_and_get_blocks_v2() {
+    let num_tests = 10;
+    let mut harness = ComponentHarness::default();
+    let mut storage = storage_fixture(&harness);
+
+    let era_id = harness.rng.gen_range(0..10).into();
+    let height = harness.rng.gen_range(0..100);
+
+    for i in 0..num_tests {
+        let block = Block::random_with_specifics(
+            &mut harness.rng,
+            era_id,
+            height + i,
+            HashingAlgorithmVersion::HASH_V2_PROTOCOL_VERSION,
+            i == num_tests - 1,
+        );
+
+        assert!(put_block(
+            &mut harness,
+            &mut storage,
+            Box::new(block.clone())
+        ));
+
+        let mut txn = storage.env.begin_ro_txn().unwrap();
+        let block_body_merkle = block.body().merklize();
+
+        for hash in [
+            *block_body_merkle
+                .deploy_hashes
+                .merkle_linked_list_node_hash(),
+            *block_body_merkle
+                .transfer_hashes
+                .merkle_linked_list_node_hash(),
+            *block_body_merkle.proposer.merkle_linked_list_node_hash(),
+        ] {
+            assert!(matches!(
+                txn.get_value::<_, [Digest; 2]>(storage.block_body_v2_db, &hash),
+                Ok(Some(_))
+            ));
+        }
+
+        assert!(matches!(
+            txn.get_value::<_, Vec<DeployHash>>(
+                storage.deploy_hashes_db,
+                block_body_merkle.deploy_hashes.value_hash()
+            ),
+            Ok(Some(_))
```

Comment:
Can we compare with the actual value here? (Also below.)

```suggestion
            Ok(Some(block.body().deploy_hashes().clone()))
```